--- fedora-collabora-kernel/KERNEL/kernel.spec.orig	2023-12-21 11:51:03.112552396 -0400
+++ fedora-collabora-kernel/KERNEL/kernel.spec	2023-12-21 11:56:23.006269055 -0400
@@ -796,7 +796,10 @@
 # exact git commit you can run
 #
 # xzcat -qq ${TARBALL} | git get-tar-commit-id
-Source0: linux-%{tarfile_release}.tar.xz
+#Source0: linux-%{tarfile_release}.tar.xz
+
+# pull from https://github.com/torvalds/linux/archive/refs/tags/
+Source0: 
 
 Source1: Makefile.rhelver
 
@@ -976,6 +979,7 @@
 %if !%{nopatches}
 
 Patch1: patch-%{patchversion}-redhat.patch
+Patch100: 0001-rk3588-commits-from-Collabora-tree.patch
 %endif
 
 # empty final patch to facilitate testing of kernel patches
@@ -1679,6 +1683,12 @@
 %endif
 
 %prep
+%if 0%{?copr_projectname:1}
+# This happens only in Copr
+pwd
+printenv
+%endif
+
 # do a few sanity-checks for --with *only builds
 %if %{with_baseonly}
 %if !%{with_up}
@@ -1741,6 +1751,7 @@
 
 %if !%{nopatches}
 
+ApplyOptionalPatch 0001-rk3588-commits-from-Collabora-tree.patch
 ApplyOptionalPatch patch-%{patchversion}-redhat.patch
 %endif
 
